sherpa/models/regrid.py

```diff
@@ -112,36 +115,163 @@ def end(self):
         of the `hi` array or of the `lo` array, depending on whether
         the axis is integrated or not, respectively.
         """
-        if self.is_ascending and self.is_integrated:
-            return self.hi[-1]
-        if self.is_ascending and not self.is_integrated:
-            return self.lo[-1]
-        if self.is_integrated:
-            return self.hi[0]
-        return self.lo[0]
+        pass
 
     @property
+    @abstractmethod
     def size(self):
         """The size of the axis."""
-        return self.lo.size
+        pass
 
     def overlaps(self, other):
         """Check if this axis overlaps with another.
 
         Parameters
         ----------
-        other : Axis
+        other : Axis instance
             The axis to compare to.
 
         Returns
         -------
         bool
             True if they overlap, False if not
         """
+        # Could apply this check but this is not expected to be
+        # called directly, so leave for now.
+        #
+        # if not isinstance(other, Axis):
+        #     raise TypeError("other argument must be an axis")
+
         num = max(0, min(self.end, other.end) - max(self.start, other.start))
         return bool(num != 0)
 
 
+class PointAxis(Axis):
+    """Represent a point (not integrated) axis of a N-D object.
+
+    The length can not be changed once set.
+
+    Parameters
+    ----------
+    x : array_like or None
+        The starting point of the axis. If `None` or `[]` then the
+        data axis is said to be empty. The axis can be in ascending or
+        descending order but this is not checked.
+
+    """
+
+    def __init__(self, x):
+        self._x = _to_readable_array(x)
+        if self._x is None:
+            return
+
+        nx = len(self._x)
+        if nx > 0:
+            self._is_ascending = self._x[-1] > self._x[0]
```

Comment:
How about `x = [3, 2, 5]`? 
How about `np.all(np.sort(x) == x)`? Thinking about it, there should probably a check in the `Axis` base class that ensures that it's monotonic (either ascending or decending). Something like 
```
d=np.diff(x)
if not np.all(d > 0) or np.all(d < 0):
    raise ...
```

Reply:
```
x = [3, 4, 4]
```

is another case. However, I didn't want to add extra checks as I'm still not completely sure about what we really support, and decided not to rock the boat too much. I should go and check a 1D data array that is unordered - which is something I think we have supported in the past, even if some of the plots may appear less-than-endearing to the user in this case.

Reply:
I've created #1577 to track this, as I do not know the intent of the original code, or once we added the "regrid" support.

Reply:
In other words, I don't want to enforce a monotonic axis before we've had time to work through all the implications?

Reply:
The case with a repeated independent axis should be handled as a specific case and I don't think we have implemented any case which would properly fit the data like  this `x = [3, 4, 4]`.

Reply:
Well, you can fit such a dataset now - the following is a very-simplified case where we fit a constant, so the independent coordinate does not matter, but technically you could fit something like gauss2d:

```
>>> from sherpa.ui import *
WARNING: imaging routines will not be available, 
failed to import sherpa.image.ds9_backend due to 
'RuntimeErr: DS9Win unusable: Could not find ds9 on your PATH'
>>> load_arrays(1, [2, 5, 2], [2, 6, 4])
>>> set_source(const1d.mdl)
>>> fit()
Dataset               = 1
Method                = levmar
Statistic             = chi2gehrels
Initial fit statistic = 2.96289
Final fit statistic   = 0.823007 at function evaluation 4
Data points           = 3
Degrees of freedom    = 2
Probability [Q-value] = 0.662653
Reduced statistic     = 0.411504
Change in statistic   = 2.13988
   mdl.c0         3.59539      +/- 1.77422     
>>> load_arrays(1, [100, 200, 300], [2, 6, 4])
>>> set_source(const1d.mdl2)
>>> fit()
Dataset               = 1
Method                = levmar
Statistic             = chi2gehrels
Initial fit statistic = 2.96289
Final fit statistic   = 0.823007 at function evaluation 4
Data points           = 3
Degrees of freedom    = 2
Probability [Q-value] = 0.662653
Reduced statistic     = 0.411504
Change in statistic   = 2.13988
   mdl2.c0        3.59539      +/- 1.77422     
```